R/shared-methods.R

```diff
@@ -173,10 +179,18 @@ DataClass <- R6::R6Class(
     #' cleaning steps be kept and output in a list.
     #' @param get Logical, defaults to FALSE. Should the class `get` method be
     #' called (this will download, clean, and process data at initialisation).
+    #' @param process_options list, additional arguments to control what
+    #' functions are called during processing. For some datasets setting these
+    #' to FALSE may cause errors, but for others improve speed.
+    #' Avaliable options are:
+    #' "calculate_columns_from_existing_data" = TRUE,
+    #' "add_extra_na_cols" = TRUE,
+    #' "set_negative_values_to_zero" = TRUE
     #' @export
     initialize = function(level = "1", regions,
                           totals = FALSE, localise = TRUE,
-                          verbose = TRUE, steps = FALSE, get = FALSE) {
+                          verbose = TRUE, steps = FALSE, get = FALSE,
+                          process_options = list()) {
```

Comment:
I would have thought the obvious thing to do is using inheritance to get a default set of processing options and then optionally update at initialisation (or some other time) if you want a different set.